### DISCUSSION TITLE
Plans for Summer 2021

### DISCUSSION THREAD

**Discussion by gingerBill** [odin-lang/Odin#918] [Planning]
(No specific order as of yet)

* [ ] Generate `core:sys/windows` library from WinMD files
* [ ] Replace (and remove) `core:sys/win32` with `core:sys/windows`
* [x] Incorporate odin formatter (odinfmt) as part of core library
* [ ] Incorporate odinfmt as part toolchain in the compiler (external DLL/shared library)
* [ ] Implement decent testing runner
* [ ] Generate other parts of the toolchain as part of the CI
* [ ] Finish and finalize `core:sync/sync2`
* [ ] Replace `core:sync` with `core:sync/sync2`

**Comment by ThisDevDane** [comment]
The more and more we do this type of things `Generate other parts of the toolchain as part of the CI`, would be increasingly nice if we had it all just running in a makefile (including windows) since that makes the CI very simple.
I've said it before but we already have a depdency on something that can build a makefile since nmake comes as part of MSVC.

**Reply by gingerBill** [reply]
I understand `Makefile` work with Windows through `nmake`, but I do still heavily prefer the `build.bat` for Windows. The `Makefile` can easily just call the `build.bat` file. 

Yes I understand having one thing to do everything can be better but for Windows, a batch file is a lot easier to deal with for other reasons.

**Comment by 3er0th** [comment]
Do you expect to release the 'Generate' source into the wild? If so, I will learn Odin in order to generate other projections.
